Frameworks/Foundation/NSOperationQueue.mm

```diff
@@ -472,109 +477,63 @@ - (void)setSuspended:(BOOL)suspend {
 /**
  @Status Interoperable
 */
-- (BOOL)isSuspended {
-    [priv->suspendedCondition lock];
-    int ret = priv->isSuspended;
-    [priv->suspendedCondition unlock];
-
-    return ret;
-}
-
-/**
- @Status Interoperable
-*/
 - (id)resume {
-    [priv->suspendedCondition lock];
-    if (priv->isSuspended) {
-        priv->isSuspended = FALSE;
-        [priv->suspendedCondition broadcast];
-    }
-    [priv->suspendedCondition unlock];
-
+    _priv->Resume();
     return self;
 }
 
 /**
  @Status Interoperable
 */
 - (id)suspend {
-    [priv->suspendedCondition lock];
-    priv->isSuspended = TRUE;
-    [priv->suspendedCondition unlock];
-
+    _priv->Suspend();
     return self;
 }
 
 /**
  @Status Interoperable
 */
 - (void)setName:(id)name {
-    id oldName = priv->_name;
-    priv->_name = [name copy];
-    [oldName release];
+    _priv->SetName(name);
 }
 
 /**
  @Status Interoperable
 */
 - (id)name {
-    if (priv->_name == nil) {
-        char szName[255];
-        sprintf_s(szName, sizeof(szName), "NSOperationQueue %08x", (unsigned int)self);
-        priv->_name = [[NSString stringWithCString:szName] retain];
-    }
+    return _priv->Name();
+}
 
-    return priv->_name;
+- (BOOL)isSuspended {
+    return _priv->IsSuspended();
 }
 
 /**
  @Status Interoperable
 */
 + (id)mainQueue {
-    if (_mainQueue == nil) {
-        _mainQueue = [[self alloc] _initMainThread];
-    }
-
+    static StrongId<NSOperationQueue> _mainQueue = [[[self alloc] init] autorelease];
```

Comment:
You're right, I overlooked the StrongId. Never mind...

---
In reply to: [95456936](https://github.com/Microsoft/WinObjC/pull/1676#discussion_r95456936) [](ancestors = 95456936)